gnosis/safe/api/transaction_service_api/transaction_service_api.py

```diff
@@ -194,7 +201,20 @@ def get_transactions(
         response = self._get_request(url)
         if not response.ok:
             raise SafeAPIException(f"Cannot get transactions: {response.content}")
-        return response.json().get("results", [])
+
+        transactions = response.json().get("results", [])
+
+        if safe_tx_hash_arg := kwargs.get("safe_tx_hash", None):
+            safe_tx_hash = HexBytes(safe_tx_hash_arg)
+            if any(
+                self._build_transaction_service_tx(tx).safe_tx_hash != safe_tx_hash
+                for tx in transactions
+            ):
+                raise SafeAPIException(
```

Comment:
Create a custom exception for this, inheriting from SafeAPIException. For example "ApiSafeTxHashNotMatchingException"